CMakeLists.txt

```diff
@@ -40,17 +50,18 @@ file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/
 # ---- Create library ----
 
 # Note: for header-only libraries change all PUBLIC flags to INTERFACE and create an interface
-# target: add_library(Greeter INTERFACE) set_target_properties(Greeter PROPERTIES
-# INTERFACE_COMPILE_FEATURES cxx_std_17)
-
+# target! EITHER: add_library(Greeter INTERFACE) OR:
 add_library(Greeter ${headers} ${sources})
 
-set_target_properties(Greeter PROPERTIES CXX_STANDARD 17)
+# EITHER: set_target_properties(Greeter PROPERTIES CXX_STANDARD 17) # OR:
+target_compile_features(Greeter PUBLIC cxx_std_17)
 
 # being a cross-platform target, we enforce standards conformance on MSVC
-target_compile_options(Greeter PUBLIC "$<$<BOOL:${MSVC}>:/permissive->")
+target_compile_options(Greeter PUBLIC "$<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/permissive>")
```

Comment:
No need to be this explicit imo. This is a C++ starter, so we can safely assume the target to be in C++ or the user to know what he's doing.

Reply:
this is recommended [here](https://discourse.cmake.org/t/is-this-really-a-better-style-to-use-generator-expressions-every-time/2753/4?u=clausklein)

Reply:
I see, hadn't thought about mingw as a consumer of a MSVC package. 👍